src/osd/scrubber/pg_scrubber.cc

```diff
@@ -1752,7 +1750,11 @@ void PgScrubber::set_queued_or_active()
 
 void PgScrubber::clear_queued_or_active()
 {
-  m_queued_or_active = false;
+  if (m_queued_or_active) {
+    m_queued_or_active = false;
+    // and just in case snap trimming was blocked by the aborted scrub
+    m_pg->snap_trimmer_scrub_complete();
+  }
 }
```

Comment:
Since this function now takes on the extra step of restarting snaptrimming, I would add a comment like the one you have in lines 195 and 211 on top of the function itself, either in this file or the .h file. Something like:

src/osd/scrubber/pg_scrubber.h
```
// Clears `m_queued_or_active` and restarts snaptrimming
 void clear_queued_or_active() final;
```

We could also consider renaming this function (maybe `clear_and_restart()`?), but I understand if this would be too disruptive to the code. Either way, we should have a comment by the function definition clarifying what it does.